### PR TITLE
feat(ODC-464): add GitSourceAnalysis API types

### PR DESCRIPTION
**Pull request by MatousJobanek**
Adds GitSourceAnalysis CRD. It will be a trigger of the build type detection and the result of the git repo analysis will be written into the status of this CRD.
When the analysis is done, then the controller sets:
* `analyzed` to `true` to easily filter out those ones that are already analyzed
* an error message to `error` if anything failed during the analysis
* result of the analysis to `BuildEnvStatistics` where the detected languages and build environment types are stored

### CODE REVIEW

**Comment by sbose78** [comment]
.. hmm this repo needs a CI.

File: pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
```
Comment (alexeykazakov) [inline pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go]: We can remove this :)

File: pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+
+	// GitSourceRef refers to the GitSource to be analyzed
+	GitSourceRef GitSourceRef `json:"gitSourceRef"`
+}
+
+// GitSourceRef refers to the GitSource to be analyzed
+type GitSourceRef struct {
+	// Name is the name of the GitSource that contains all necessary information of the git repo
```
Comment (alexeykazakov) [inline pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go]: Worth mentioning that it's the name of the GitSource **in the same namespace** unless you want to be able to refer to GitSources from another namespace. But in that case we would need to add `namespace` field too.

File: pkg/apis/devconsole/v1alpha1/zz_generated.openapi.go [hunk 3]
```diff
@@ -13,9 +13,9 @@ import (
 
 func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
 	return map[string]common.OpenAPIDefinition{
-		"github.com/redhat-developer/devconsole-api/pkg/apis/devconsole/v1alpha1.GitSource":       schema_pkg_apis_devconsole_v1alpha1_GitSource(ref),
-		"github.com/redhat-developer/devconsole-api/pkg/apis/devconsole/v1alpha1.GitSourceSpec":   schema_pkg_apis_devconsole_v1alpha1_GitSourceSpec(ref),
-		"github.com/redhat-developer/devconsole-api/pkg/apis/devconsole/v1alpha1.GitSourceStatus": schema_pkg_apis_devconsole_v1alpha1_GitSourceStatus(ref),
+		"github.com/redhat-developer/git-service/pkg/apis/devconsole/v1alpha1.GitSource":       schema_pkg_apis_devconsole_v1alpha1_GitSource(ref),
```
Comment (alexeykazakov) [inline pkg/apis/devconsole/v1alpha1/zz_generated.openapi.go]: Wrong package.

File: pkg/apis/devconsole/v1alpha1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -25,126 +25,177 @@ import (
...
+// DeepCopyObject is an autogenerated deepcopy function, copying the receiver, creating a new runtime.Object.
+func (in *GitSourceAnalysisList) DeepCopyObject() runtime.Object {
+	if c := in.DeepCopy(); c != nil {
+		return c
+	}
+	return nil
+}
+
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
-func (in *ComponentStatus) DeepCopyInto(out *ComponentStatus) {
```
Comment (alexeykazakov) [inline pkg/apis/devconsole/v1alpha1/zz_generated.deepcopy.go]: Same as in the other PR. Make sure you are not deleting Component API.

**Review by alexeykazakov — COMMENTED** [review]
Looks good! Just a minor description comment and also please take a look at the generated code.

File: pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go (hunk 1 above)
Comment (MatousJobanek) [reply pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go]: removed in 57787c6

File: pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go (hunk 2 above)
Comment (MatousJobanek) [reply pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go]: good point. changed https://github.com/redhat-developer/devconsole-api/pull/8/files#diff-40860617e11838e34856b8f57fc5cda7R20
thanks

File: pkg/apis/devconsole/v1alpha1/zz_generated.openapi.go (hunk 3 above)
Comment (MatousJobanek) [reply pkg/apis/devconsole/v1alpha1/zz_generated.openapi.go]: Same as in the other PR - this change has disappeared after generating it in `devconsole-operator` repo

File: pkg/apis/devconsole/v1alpha1/zz_generated.deepcopy.go (hunk 4 above)
Comment (MatousJobanek) [reply pkg/apis/devconsole/v1alpha1/zz_generated.deepcopy.go]: fixed in 87977b1

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**